app/presenters/scoped_search_results_presenter.rb

```diff
@@ -0,0 +1,58 @@
+class ScopedSearchResultsPresenter < SearchResultsPresenter
+
+  def to_hash
+    super.merge({
+      is_scoped?: true,
+      scope_title: scope_title,
+      unscoped_results_any?: unscoped_results.any?,
+      unscoped_result_count: result_count_string(unscoped_result_count),
+      results: presentable_result_list,
+    })
+  end
+
+private
+
+  attr_reader :unscoped_results
+
+  def filter_fields
+    []
+  end
+
+  def scope_title
+    search_response["scope"]["title"]
+  end
+
+  def presentable_result_list
```

Comment:
Yes I like this, it is clearer. Thank you for explaining it clearly here. Very useful.